### PR TITLE
OHRI-2090 OHRI-2094 fix tb summary

### PR DESCRIPTION
**Pull request by kajambiya**
## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes the ticket number in the format `OHRI-123 My PR title`.
- [ ] My work includes tests or is validated by existing tests.

## Summary
This PR fixes the issue of some values not displaying on the TB Summary page. 
## Screenshots
Screenshot 2024-01-15 at 12 29 31

### CODE REVIEW

File: packages/esm-tb-app/src/views/patient-summary/tb-patient-summary.component.tsx [hunk 1]
```diff
@@ -124,7 +130,12 @@ const TBSummaryOverviewList: React.FC<PatientChartProps> = ({ patientUuid }) =>
         key: 'regimen',
         header: t('regimen', 'Regimen'),
         getValue: (encounter) => {
-          return getObsFromEncounter(encounter, config.obsConcepts.regimen);
+          const tBEnrollmentType = findObs(encounter, config.obsConcepts.tBEnrollmentType)?.value?.uuid;
```
Comment (eudson) [inline packages/esm-tb-app/src/views/patient-summary/tb-patient-summary.component.tsx:133]: Can you encapsulate the logic of getting the regimen in a function and replace in both occurences?

**Review by eudson — CHANGES_REQUESTED** [review]

File: packages/esm-tb-app/src/views/patient-summary/tb-patient-summary.component.tsx [hunk 2]
```diff
@@ -42,47 +43,52 @@ const TBSummaryOverviewList: React.FC<PatientChartProps> = ({ patientUuid }) =>
         key: 'type',
         header: t('type', 'Type'),
         encounterTypes: [config.encounterTypes.tbProgramEnrollment],
-        getObsValue: (encounter) => {
+        getObsValue: ([encounter]) => {
```
Comment (pirupius) [inline packages/esm-tb-app/src/views/patient-summary/tb-patient-summary.component.tsx:46]: Was this formatting intentional?

File: packages/esm-tb-app/src/views/patient-summary/tb-patient-summary.component.tsx [hunk 3]
```diff
@@ -42,47 +43,52 @@ const TBSummaryOverviewList: React.FC<PatientChartProps> = ({ patientUuid }) =>
...
         header: t('regimen', 'Regimen'),
         encounterTypes: [config.encounterTypes.tbProgramEnrollment],
-        getObsValue: (encounter) => {
-          return getObsFromEncounter(encounter, config.obsConcepts.regimen);
+        getObsValue: ([encounter]) => {
+          const tBEnrollmentType = findObs(encounter, config.obsConcepts.tBEnrollmentType)?.value?.uuid;
+          const regimen =
+            tBEnrollmentType === config.obsConcepts.dsTBEnrollment
+              ? config.obsConcepts.dSregimen
+              : config.obsConcepts.dRregimen;
```
Comment (pirupius) [inline packages/esm-tb-app/src/views/patient-summary/tb-patient-summary.component.tsx:75]: similar comment about making this reusable

File: packages/esm-tb-app/src/views/patient-summary/tb-patient-summary.component.tsx (hunk 2 above)
Comment (kajambiya) [reply packages/esm-tb-app/src/views/patient-summary/tb-patient-summary.component.tsx:47]: Yah, the `getObsValue` function from the `SummaryCard` component expects an array of encounters.

File: packages/esm-tb-app/src/views/patient-summary/tb-patient-summary.component.tsx (hunk 1 above)
Comment (kajambiya) [reply packages/esm-tb-app/src/views/patient-summary/tb-patient-summary.component.tsx:130]: Done

File: packages/esm-tb-app/src/views/patient-summary/tb-patient-summary.component.tsx (hunk 3 above)
Comment (kajambiya) [reply packages/esm-tb-app/src/views/patient-summary/tb-patient-summary.component.tsx]: Done

**Review by pirupius — APPROVED** [review]

**Review by eudson — APPROVED** [review]
LGTM

**Pull request merged**